brave/src/test/java/com/linecorp/armeria/server/brave/BraveServiceIntegrationTest.java

```diff
@@ -65,34 +82,123 @@ protected void init() {
             return HttpResponse.of(HttpStatus.NOT_FOUND);
         });
         sb.service("/foo", (ctx, req) -> HttpResponse.of(OK, MediaType.PLAIN_TEXT_UTF_8, "bar"));
-        sb.service("/extra",
+        sb.service("/async", (ctx, req) -> asyncResponse(future ->
+            future.complete(HttpResponse.of(OK, MediaType.PLAIN_TEXT_UTF_8, "bar"))));
+
+        sb.service("/exception", (ctx, req) -> {
+            // TODO: HttpResponseException.of(status, "not ready")
+            throw HttpResponseException.of(HttpStatus.SERVICE_UNAVAILABLE);
+        });
+        sb.service("/exceptionAsync", (ctx, req) -> asyncResponse(future ->
+            // TODO: HttpResponseException.of(status, "not ready")
+            future.completeExceptionally(HttpResponseException.of(HttpStatus.SERVICE_UNAVAILABLE))));
+
+        sb.service("/items/:itemId",
                    (ctx, req) -> HttpResponse.of(OK, MediaType.PLAIN_TEXT_UTF_8,
-                                                 String.valueOf(req.headers().get(EXTRA_KEY))));
+                                                 String.valueOf(ctx.pathParam("itemId"))));
+        sb.service("/async_items/:itemId", (ctx, req) -> asyncResponse(future ->
+            future.complete(HttpResponse.of(OK, MediaType.PLAIN_TEXT_UTF_8,
+                                                String.valueOf(ctx.pathParam("itemId"))))));
+        // TODO: how do we mount "/items/:itemId" under the prefix "/nested"?
+
+        // TODO: these fail because the context isn't here for some reason?
+        sb.service("/child", (ctx, req) -> {
```

Comment:
anything obvious here? the body is not in the server context.

Reply:
I mean this lambda cannot see the trace context... neither does the one for baggage

Reply:
Actually I'm surprised this doesn't fail, I thought we have some validation for duplicate paths /cc @minwoox @ikhoon 

While I'm not sure, I suspect the issue is having two services mounted at the same path `/child`, which I think is messing with the decoration logic.

Reply:
>  I thought we have some validation for duplicate paths

The default is just warning. https://github.com/line/armeria/blob/master/core/src/main/java/com/linecorp/armeria/server/ServerBuilder.java#L199
We should set another handler for different behavior.

Reply:
Ah thanks now I remember, probably was a warning here then :)

Reply:
this test is a single child household. the other will be sold off for medical experiments

Reply:
this duplicate made the other things fail!